### PR TITLE
feat: add monitoring and logging

### PR DESCRIPTION
**Pull request by zachariahmiller**
Add promtail, fix netpols, add prometheus to standard-package, add grafana

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: capabilities/loki/chart/templates/loki-dashboards.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ items:
   kind: ConfigMap
   metadata:
     name: {{ printf "%s-%s" (include "loki.name" $) $dashboardName | trunc 63 | trimSuffix "-" }}
-    namespace: monitoring
+    namespace: grafana
     labels:
       grafana_dashboard: "1"
   data:
```
Comment (mjnagel) [inline capabilities/loki/chart/templates/loki-dashboards.yaml]: The second half of this file isn't being used and could be removed (everything after the yaml separator on line 13).

File: capabilities/loki/chart/templates/loki-dashboards.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ apiVersion: v1
 kind: ConfigMap
```
Comment (mjnagel) [inline capabilities/loki/chart/templates/loki-dashboards.yaml:2]: Can't comment on exact line (2) but you can remove `{{- $files := .Files.Glob "src/dashboards/*.json" }}` - that's not used.

File: capabilities/neuvector/zarf.yaml [hunk 3]
```diff
@@ -37,9 +41,6 @@ components:
...
 
-  - name: uds-neuvector-config
-    charts:
-      - name: uds-neuvector-config
-        namespace: neuvector
-        version: 0.1.0
-        localPath: chart
+
+
+
```
Comment (mjnagel) [inline capabilities/neuvector/zarf.yaml]:
```suggestion
```

File: capabilities/promtail/values/values.yaml [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    rules: []
+  relabelings: []
+  scheme: https
+  scrapeTimeout: null
+  targetLabels: []
+  tlsConfig:
+    caFile: /etc/prom-certs/root-cert.pem
+    certFile: /etc/prom-certs/cert-chain.pem
+    insecureSkipVerify: true
+    keyFile: /etc/prom-certs/key.pem
```
Comment (mjnagel) [inline capabilities/promtail/values/values.yaml:157]: Just dropping as a note, the specific tlsConfig here should definitely be a part of a broader pepr capability - https://github.com/defenseunicorns/uds-core/issues/17

File: capabilities/promtail/values/values.yaml [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+#   clients:
+#     {{- if eq .Values.loki.strategy "monolith" }}
+#     - url: http://logging-loki.logging.svc.cluster.local:3100/loki/api/v1/push
+#     {{- else }}
+#     - url: http://logging-loki-write.logging.svc.cluster.local:3100/loki/api/v1/push
+#     {{- end }}
+
+  snippets:
+    scrapeConfigs: >
+      - job_name: kubernetes-pods
```
Comment (mjnagel) [inline capabilities/promtail/values/values.yaml:7]: I think we can nuke most of this except for the node log stuff (journal/systemd), and instead add those pieces via `snippets.extraScrapeConfigs`.

Comment (zachariahmiller) [reply capabilities/promtail/values/values.yaml:7]: I created an issue for this

File: capabilities/grafana/chart/templates/datasources.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          tags:
+          - job
+          - instance
+          - pod
+          - namespace
+      name: Tempo
+      orgId: 1
+      type: tempo
+      uid: tempo
+      url: http://tempo.tempo.svc:3100
```
Comment (mjnagel) [inline capabilities/grafana/chart/templates/datasources.yaml]: Probably should leave this the tempo datasource off until we decide on adding tempo or not?

File: capabilities/grafana/zarf.yaml [hunk 7]
```diff
@@ -15,13 +15,14 @@ components:
...
       - name: uds-grafana-config
         namespace: grafana
         version: 0.1.0
         localPath: chart
+    images:
+      - docker.io/grafana/grafana:10.1.5
+      - docker.io/curlimages/curl:7.85.0
+      - docker.io/grafana/grafana-image-renderer:latest
+      - docker.io/library/busybox:1.31.1
+      - docker.io/bats/bats:v1.4.1
```
Comment (mjnagel) [inline capabilities/grafana/zarf.yaml]:
```suggestion
      - docker.io/library/busybox:1.31.1
```
I don't think we use either of these two - image renderer is disabled and the bats image is for their test framework.

File: capabilities/loki/chart/values.yaml [hunk 8]
```diff
@@ -1,2 +1,2 @@
 grafana:
-  enabled: false
+  enabled: true
```
Comment (mjnagel) [inline capabilities/loki/chart/values.yaml]: nit: we could probably remove this value (along with the conditional wrapper around https://github.com/defenseunicorns/uds-core/blob/add_monitoring_and_logging/capabilities/loki/chart/templates/loki-dashboards.yaml). I don't think it provides a ton of value since the configmap won't break anything if deployed without grafana, and we don't want there to be any cases where grafana isn't deployed in our bundle.

File: capabilities/prometheus-stack/values/values.yaml [hunk 9]
```diff
@@ -120,19 +30,8 @@ kube-state-metrics:
     #registry: registry1.dso.mil
     #repository: ironbank/opensource/kubernetes/kube-state-metrics
     #tag: v2.10.0
-  podAnnotations:
-    #PEPR?
-    bigbang.dev/istioVersion: 1.19.0-enterprise
   prometheus:
     monitor:
```
Comment (mjnagel) [inline capabilities/prometheus-stack/values/values.yaml]: nit: should delete this so that it doesn't null the defaults out.
```suggestion
```

File: capabilities/prometheus-stack/values/values.yaml [hunk 10]
```diff
@@ -157,12 +56,6 @@ kubeStateMetrics:
...
       enabled: false
-#NEEDED? (begin)
-#loki:
-#  enabled: true
-#minioOperator:
-#  enabled: false
-#NEEDED? (end)
 nameOverride: kube-prometheus-stack
 networkPolicies:
   controlPlaneCidr: 0.0.0.0/0
```
Comment (mjnagel) [inline capabilities/prometheus-stack/values/values.yaml]: nit: this is a carry over from BB, along with the 3 lines after.

File: capabilities/promtail/chart/templates/istio/peer-authentication.yaml [hunk 11]
```diff
@@ -1,7 +1,7 @@
 apiVersion: security.istio.io/v1beta1
```
Comment (mjnagel) [inline capabilities/promtail/chart/templates/istio/peer-authentication.yaml]: nit: this is redundant since the istio-system one applies globally.

File: capabilities/promtail/values/values.yaml [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+  - mountPath: /var/log
+    name: varlog
+    readOnly: true
+  - mountPath: /etc/machine-id
+    name: machine-id
+    readOnly: true
+    subPath: machine-id
+
+hostname: null
+
```
Comment (mjnagel) [inline capabilities/promtail/values/values.yaml]:
```suggestion
```
nit: BB carryover

File: capabilities/promtail/zarf.yaml [hunk 13]
```diff
@@ -1,11 +1,28 @@
...
+        gitPath: charts/promtail
+        valuesFiles:
+          - values/values.yaml
+      - name: uds-promtail-config
+        namespace: promtail
+        version: 0.1.0
+        localPath: chart
+    images:
+      - docker.io/jimmidyson/configmap-reload:v0.8.0
+      - docker.io/busybox:1.33
```
Comment (mjnagel) [inline capabilities/promtail/zarf.yaml]:
```suggestion
```
nit: don't think busybox is used?

**Review by mjnagel — COMMENTED** [review]
Mostly some nits on extra things I noticed when scrolling through. 2 other broad comments probably to track as follow on issues:
- Some of these charts have new versions available upstream vs what is here
- Image stuff...some of the values we have for repo/image aren't really needed but pending flavor discussions might be?

File: capabilities/grafana/chart/templates/datasources.yaml (hunk 6 above)
Comment (zachariahmiller) [reply capabilities/grafana/chart/templates/datasources.yaml]: done

File: capabilities/grafana/validate.yaml [hunk 14]
```diff
@@ -3,5 +3,8 @@ tasks:
     actions:
       - description: Validate...
         cmd: "echo Replace Me"
+
         # wait:
-        #   cluster:
+        #   network:
```
Comment (jeff-mccoy) [inline capabilities/grafana/validate.yaml]: for now we've just been adding a healthcheck to some k8s resource, we can do the network tests in another PR

File: capabilities/neuvector/zarf.yaml [hunk 15]
```diff
@@ -20,26 +20,22 @@ components:
...
-          - values/neuvector-minimal-values.yaml
+          - values/values.yaml
       - name: monitor
         url: https://neuvector.github.io/neuvector-helm/
         version: 2.6.4
         namespace: neuvector
         gitPath: charts/monitor
         valuesFiles:
           - values/neuvector-monitor-values.yaml
+      - name: uds-neuvector-config
```
Comment (jeff-mccoy) [inline capabilities/neuvector/zarf.yaml:31]: not sure how I feel about combining the app portion with config things in one component. Realize it saves an extra import line, but don't know that this is necessarily the right move.

Comment (zachariahmiller) [reply capabilities/neuvector/zarf.yaml:31]: I dont have a problem changing it back in each place, but at least in the current state it didnt seem like we are gaining anything from the config charts being separate except more import statements

File: capabilities/promtail/values/values.yaml [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+    name: varlog
+    readOnly: true
+  - mountPath: /etc/machine-id
+    name: machine-id
+    readOnly: true
+    subPath: machine-id
+
+resources:
+  limits:
+    cpu: 500m
```
Comment (jeff-mccoy) [inline capabilities/promtail/values/values.yaml:130]: these limits seem a bit .... low? Are we sure this will be fine long-term?

File: capabilities/grafana/validate.yaml (hunk 14 above)
Comment (zachariahmiller) [reply capabilities/grafana/validate.yaml]: Do you want me to add the resource checks in the validate for each of the modified apps now? I can do that.

File: capabilities/promtail/zarf.yaml [hunk 17]
```diff
@@ -1,11 +1,27 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/defenseunicorns/zarf/main/zarf.schema.json
```
Comment (jeff-mccoy) [inline capabilities/promtail/zarf.yaml]: we don't put these tags in zarf.yaml anymore as they are loaded via the project root config in vscode

File: capabilities/promtail/zarf.yaml [hunk 18]
```diff
@@ -1,11 +1,27 @@
...
+    description: "Deploy Promtail"
+    charts:
+      - name: promtail
+        url: https://grafana.github.io/helm-charts/
+        version: 6.15.3
+        namespace: promtail
+        gitPath: charts/promtail
+        valuesFiles:
+          - values/values.yaml
+      - name: uds-promtail-config
```
Comment (jeff-mccoy) [inline capabilities/promtail/zarf.yaml:19]: same comment regarding app vs config as their own components. maybe it's fine--but we need to decided and try to stick to a common pattern if we can I think

File: release-please-config.json [hunk 19]
```diff
@@ -16,8 +16,13 @@
         "packages/standard/zarf.yaml",
         "bundles/k3d-istio/uds-bundle.yaml",
         "bundles/k3d-standard/uds-bundle.yaml",
+<<<<<<< HEAD
```
Comment (jeff-mccoy) [inline release-please-config.json]: merge conflicts

**Review by jeff-mccoy — DISMISSED** [review]
I didnt take a super hard look at the netpols since I'm working on that separately right now.

File: capabilities/promtail/values/values.yaml (hunk 16 above)
Comment (zachariahmiller) [reply capabilities/promtail/values/values.yaml:137]: definitely not sure. That is what is set in the dubbd k3d deployment. I personally really like the idea of trying to find a better way to dynamically handle resource allocation with vpa or whatever else when it becomes an issue

File: capabilities/promtail/zarf.yaml (hunk 17 above)
Comment (zachariahmiller) [reply capabilities/promtail/zarf.yaml]: no problem i already had them there i can remove

File: capabilities/promtail/values/values.yaml (hunk 16 above)
Comment (zachariahmiller) [reply capabilities/promtail/values/values.yaml:130]: definitely not sure. That is what is set in the dubbd k3d deployment. I personally really like the idea of trying to find a better way to dynamically handle resource allocation with vpa or whatever else when it becomes an issue

File: capabilities/neuvector/zarf.yaml (hunk 15 above)
Comment (mjnagel) [reply capabilities/neuvector/zarf.yaml:31]: In general I like combining the config chart + app into a single component for simplicity of imports, but happy to hear compelling reasons on why that might not be a good option.

File: capabilities/promtail/values/values.yaml (hunk 16 above)
Comment (jeff-mccoy) [reply capabilities/promtail/values/values.yaml:130]: I think to be safe we should just set higher limits to avoid a needless pod termination doom-loop.

File: capabilities/grafana/validate.yaml (hunk 14 above)
Comment (jeff-mccoy) [reply capabilities/grafana/validate.yaml]: yes plz

File: capabilities/neuvector/zarf.yaml (hunk 15 above)
Comment (jeff-mccoy) [reply capabilities/neuvector/zarf.yaml:31]: yeah it might be fine, idk--worth just deciding one way and moving on. For Istio for now, we have to keep things separate because of an outstanding helm bug. But that's not a reason to force everything else that way. Originally, components were designed to logically group apps/behavior--so you could argue that one component is reasonable in that regard. It certainly would help the imports here.

Comment (zachariahmiller) [reply capabilities/neuvector/zarf.yaml:31]: per several discussions, leaving the config-charts as part of the main components

File: capabilities/grafana/validate.yaml (hunk 14 above)
Comment (zachariahmiller) [reply capabilities/grafana/validate.yaml]: threw in some checks on the pods. maybe should have used the deployments, stateful sets, etc instead. Assuming it makes sense can flesh out more/better checks in the future.

File: capabilities/promtail/values/values.yaml (hunk 16 above)
Comment (zachariahmiller) [reply capabilities/promtail/values/values.yaml:130]: i increased the limits for now, without data to back it though it feels pretty arbitrary.

**Comment by zachariahmiller** [comment]
assuming the ci passes this should be good to go

**Review by mjnagel — DISMISSED** [review]
LGTM

**Comment by jeff-mccoy** [comment]
Added the debug stuff because something looked like it was hanging and this time it passed, probably something we need to wait and see if we can reproduce with debug logs next time. 

Screenshot 2023-11-16 at 12 16 12 AM
Screenshot 2023-11-16 at 12 15 40 AM

**Review by jeff-mccoy — APPROVED** [review]

**Review by Noxsios — APPROVED** [review]
lgtm
